### PR TITLE
[wasm] Modify workload to pick threading runtime packs

### PR DESCRIPTION
**Pull request by steveisok**
This change adds the 2 wasm threading runtime packs to the wasm workload. In order for a threading runtime pack to be chosen, WorkloadManifest.targets is also modified to override the runtime pack name when the following props are set:

WasmEnableThreads - full threading support and will load `Microsoft.NETCore.App.Runtime.multithread.Mono.browser-wasm`

WasmEnablePerfTrace - runtime only threading support and will load `Microsoft.NETCore.App.Runtime.perftrace.Mono.browser-wasm`

Contributes to #72872

### CODE REVIEW

**Comment by lewing** [comment]
Need to make sure WBT knows about the packs

**Review by lewing — COMMENTED** [review]
A couple of questions but should be looks good otherwise

File: src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in [hunk 1]
```diff
@@ -93,6 +93,9 @@
     <ItemGroup Condition="'$(_MonoWorkloadTargetsMobile)' == 'true'">
       <KnownRuntimePack Update="@(KnownRuntimePack)">
         <LatestRuntimeFrameworkVersion Condition="'%(KnownRuntimePack.TargetFramework)' == '${NetCoreAppCurrent}' and '%(KnownRuntimePack.RuntimePackLabels)' == 'Mono'">**FromWorkload**</LatestRuntimeFrameworkVersion>
+        <!-- Overrides for wasm threading support -->
+        <RuntimePackNamePatterns Condition="'$(RuntimeIdentifier)' == 'browser-wasm' and '$(WasmEnableThreading)' == 'true'">Microsoft.NETCore.App.Runtime.Mono.multithread.**RID**</RuntimePackNamePatterns>
+        <RuntimePackNamePatterns Condition="'$(RuntimeIdentifier)' == 'browser-wasm' and '$(WasmEnablePerfTrace)' == 'true'">Microsoft.NETCore.App.Runtime.Mono.perftrace.**RID**</RuntimePackNamePatterns>
```
[the remark this answers is not in the document]
Comment (steveisok) [reply src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in:98]: Perftrace is single threaded userspace, multi-threaded runtime.  Do you think we should error if both are true?

**Review by lewing — APPROVED** [review]

**Comment by radical** [comment]
Looks good. I can add a test in WBT.

**Comment by radical** [comment]
`Workload installation failed: microsoft.netcore.app.runtime.mono.multithread.browser-wasm::7.0.0-ci is not found in NuGet feeds`. I don't see the this nuget being generated in the [build](https://dev.azure.com/dnceng/public/_build/results?buildId=1890352&view=logs&jobId=6f051c1a-c3e7-5cde-411e-64125624a208&j=6f051c1a-c3e7-5cde-411e-64125624a208&t=b731ef43-7f67-5a47-0aca-433fa8d3212b).

**Comment by radical** [comment]
> To get this to work, I think WasmBuildTests is going to need to wait for the threading builds to complete and then import its runtime packs.

Ok, I will add that.

**Comment by radical** [comment]
I have pushed some changes that will unblock this PR. Essentially, we copy the single runtime pack nuget to the other nuget filenames, and should allow running Wasm.Build.Tests . Once I have the PR, to get the other correct runtime packs on CI, completed, I will change this behavior to be applicable only for local builds.

It will emit this when it copies:
```
        ********************

        Note: Could not find the expected three runtime packs in /workspaces/runtime/artifacts/packages/Release/Shipping/. Found Microsoft.NETCore.App.Runtime.Mono.browser-wasm.7.0.0-dev, Microsoft.NETCore.App.Runtime.Mono.multith
read.browser-wasm.7.0.0-dev, Microsoft.NETCore.App.Runtime.Mono.perftrace.browser-wasm.7.0.0-dev .
              To support local builds, /workspaces/runtime/artifacts/packages/Release/Shipping/Microsoft.NETCore.App.Runtime.Mono.browser-wasm.7.0.0-dev.nupkg was copied as the missing nugets.

        *******************
```

**Review by radical — APPROVED** [review]

**Comment by radical** [comment]
The coreclr failure seems to be https://github.com/dotnet/runtime/issues/68289 and unrelated to this PR.
The WBT failures are known (https://github.com/dotnet/runtime/issues/72880), and unrelated.

**Pull request merged**